app/api/routers/breeze_buddy/telephony/callbacks/handlers.py

```
@@ -182,28 +189,65 @@ async def handle_callback_status(request: Request, provider: str) -> Response:
-        logger.info(f"Call with SID {call_sid} failed with status: {call_status}")
-        # Convert to string for the handler
-        if call_sid and isinstance(call_sid, str):
-            await handle_unanswered_calls(call_sid)
+                router_client = get_smart_router_client()
+                await router_client.release_pod(
+                    call_sid=str(call_sid), reason=f"status_{call_status}"
+                )
+                logger.debug(
+                    f"Smart Router release called for {call_sid} via status callback"
+                )
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)